ui/src/middlewares/searchDispatcher.js

```diff
@@ -1,68 +1,96 @@
 }
 
+function hasQueryChangedExceptSortOrPape(prevQuery, nextQuery) {
+  return !shallowEqual(
+    omit(prevQuery.toJS(), ['sort', 'page']),
+    omit(nextQuery.toJS(), ['sort', 'page'])
+  );
+}
+
+// TODO: this could be moved to the action creator
+// only advantage is that it's easier to test SEARCH_QUERY_UPDATE in search container components
+// rather than an async action
 export default function({ getState, dispatch }) {
   return next => action => {
-    if (
-      action.type === LOCATION_CHANGE &&
-      action.payload &&
-      isSearchPage(action.payload.location)
-    ) {
-      const currentLocation = getLocationFromRootState(getState());
+    if (action.type === SEARCH_QUERY_UPDATE) {
+      const { namespace } = action.payload;
+      const prevQuery = getQueryFromState(namespace, getState());
       const result = next(action);
-      const nextLocation = action.payload.location; // use this instead of `getLocationFromRootState(getState())` for testability.
-      const { isFirstRendering } = action.payload;
-      const hasPathnameChanged =
-        nextLocation.pathname !== currentLocation.pathname;
-      const hasSearchChanged = nextLocation.search !== currentLocation.search;
-      if (isFirstRendering || hasPathnameChanged || hasSearchChanged) {
-        dispatch(searchForCurrentLocation());
+      const nextState = getState();
+      const nextQuery = getQueryFromState(namespace, nextState);
+      const baseQuery = getBaseQueryFromState(namespace, nextState);
+
+      const hasQueryChanged = prevQuery !== nextQuery;
+      // TODO: better name (this is basically to trigger search when initial location sth like /literature)
+      const isInitialQueryUpdateEmpty = nextQuery === baseQuery;
+
+      if (hasQueryChanged || isInitialQueryUpdateEmpty) {
+        dispatch(searchForCurrentQuery(namespace));
       }
-      // TODO: change the shallowEqual once we make router immutable. This works for now because we generate the query from scratch everytime
-      const hasQueryChangedExceptSortOrPage = !shallowEqual(
-        omit(currentLocation.query, ['sort', 'page']),
-        omit(nextLocation.query, ['sort', 'page'])
-      );
+
       if (
-        hasQueryChangedExceptSortOrPage ||
-        hasPathnameChanged ||
-        isFirstRendering
+        hasQueryChangedExceptSortOrPape(prevQuery, nextQuery) ||
+        isInitialQueryUpdateEmpty
       ) {
-        if (nextLocation.pathname.startsWith(LITERATURE)) {
-          dispatch(fetchSearchAggregationsForCurrentLocation());
-        }
-        if (
-          nextLocation.pathname.startsWith(JOBS) &&
-          (hasPathnameChanged || isFirstRendering)
-        ) {
-          dispatch(fetchSearchAggregationsForCurrentLocation(false));
-        }
+        dispatch(fetchSearchAggregationsForCurrentQuery(namespace));
       }
       return result;
     }
+
+    if (action.type === SEARCH_BASE_QUERIES_UPDATE) {
```

Comment:
what use cases do we have for the ```SEARCH_BASE_QUERIES_UPDATE```? Is it only used for displaying 100 results for catalogers and adding the author filter?

Reply:
it's for embedded searches only, so far we have only one.